### ISSUE TITLE
`portablectl attach` does not copy foobar.target

### ISSUE THREAD

**Issue by ascent12** [systemd/systemd#10064]
**systemd version the issue has been seen with**
> v239

**Used distribution**
 > Arch Linux

**Expected behaviour you didn't see**
 > foobar.target added to the host machine

**Unexpected behaviour you saw**
 > foobar.target not added to the host machine, but **some** other services added successfully.
>`Failed to attach image: No such file or directory` output by `portablectl attach`

**Steps to reproduce the problem**
Using this striped down portable service (just contains the service files I'm using and an os-release): 
[email.tar.gz](https://github.com/systemd/systemd/files/2372929/email.tar.gz)

Verify that it's a portable service
```
# tar xzf email.tar.gz
# /usr/lib/systemd/portablectl inspect ./email
(Matching unit files with prefix 'email'.)
Image:
        /tmp/email
Portable Service:
        n/a
Operating System:
        Arch Linux
Unit files:
        email-dovecot.service
        email-opendkim.service
        email-postfix.service
        email-postgrey.service
        email.target
```

Try attach portable service
```
# /usr/lib/systemd/portablectl attach ./email
(Matching unit files with prefix 'email'.)
Failed to attach image: No such file or directory
```

View installed services
```
# systemctl list-unit-files | grep email
email-dovecot.service                        static   
email-opendkim.service                       static   
email-postfix.service                        static
```

It works correctly and all unit files are installed when email.target is removed.

**Comment by poettering** [comment]
Hmm, I tried to reproduce this here with your tarball, but it works correctly here... I am a bit puzzled...

Any chance you strace portabled while you execute this operation? i.e. something like this: 

```
systemctl start systemd-portabled && sudo strace -f -s 500 -o portabled-strace.log -p `systemctl show --value -p MainPID systemd-portabled`
```

And then let me know what this generates (paste portabled-strace.log somewhere), maybe we can see easily where the ENOENT comes from in that output.

**Comment by ascent12** [comment]
strace output: [portabled-strace.log](https://github.com/systemd/systemd/files/2457265/portabled-strace.log)

debug log: [portabled-debug.log](https://github.com/systemd/systemd/files/2457269/portabled-debug.log)

Looking at the strace myself,
```
sendmsg(3, {msg_name=NULL, msg_namelen=0, msg_iov=[{iov_base="PRIORITY=7\nSYSLOG_FACILITY=3\nCODE_FILE=../systemd-stable/src/portable/portable.c\nCODE_LINE=786\nCODE_FUNC=install_profile_dropin\nERRNO=2\nSYSLOG_IDENTIFIER=systemd-portabled\n", iov_len=172}, {iov_base="MESSAGE=", iov_len=8}, {iov_base="Profile 'default' is not accessible: No such file or directory", iov_len=62}, {iov_base="\n", iov_len=1}], msg_iovlen=4, msg_controllen=0, msg_flags=0}, MSG_NOSIGNAL) = 243
```
which refers to https://github.com/systemd/systemd/blob/v239/src/portable/portable.c#L786
This is supposed to be checking against -ENOENT instead of just ENOENT, and since /usr/lib/systemd/portable/profile/default/target.conf doesn't exist, this error would happen. It appears to be fixed in 99c89da02d, so I guess this is only an issue in v239 and is now fine in master/v240.
